components/ai_chat/core/browser/engine/engine_consumer_claude.cc

```diff
@@ -185,7 +185,9 @@ void EngineConsumerClaudeRemote::GenerateRewriteSuggestion(
     GenerationDataCallback received_callback,
     GenerationCompletedCallback completed_callback) {
   SanitizeInput(text);
-  const std::string& truncated_text = text.substr(0, max_page_content_length_);
+
+  std::string truncated_text;
+  base::TruncateUTF8ToByteSize(text, max_page_content_length_, &truncated_text);
```

Comment:
can we move these calling to `EngineConsumer::TruncateText` and then write a unit test for it using `EngineConsumer::SetMaxPageContentLengthForTesting`?

Reply:
signature would be like 
`void EngineConsumer::TruncateText(const std::string& text, const std::optional<size_t> selected_text_size, std::string* output)`